rules/ast/ast.go

```diff
@@ -732,7 +743,10 @@ func evalVectorBinop(opType BinOpType,
 	case And:
 		return lhs, true
 	case Or:
-		return lhs, true // TODO: implement OR
+		if lhs == 0 {
```

Comment:
I've very confused then - why is Or checking if the value is 0?

Or is a logical operator, it cares about labelsets not values.